### PR TITLE
WIP: WAB Article

### PR DESCRIPTION
**Pull request by atribolet**
WAB article edits 
------------------------

### CODE REVIEW

File: _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md [hunk 1]
```diff
@@ -9,18 +9,27 @@ categories:
 tags: []
 ---
 
-Recently, I've worked on a [project](https://github.com/agrc/wfrc-wab-widgets/) building custom Web AppBuilder (WAB) widgets for a client. This has been an interesting digression from our typical development projects. Working within the confines of WAB has proved to be both liberating and frustrating. Being able to focus on a single widget and not have to worry about an entire application has been great. I've loved letting Esri take care of all of the basic application stuff such as loading the application framework, web map and widgets. However, not having control over the entire system has created some headaches. In this post, I'd like to cover some solutions that have helped me.
+Recently, I have been working on a [project](https://github.com/agrc/wfrc-wab-widgets/) building custom Web AppBuilder (WAB) widgets for a client. This has been an interesting digression from our typical development projects, and working within the confines of WAB has proved to be both liberating and frustrating. In this post, I would like to share a few solutions that have helped me make the most of working with WAB.
+
+## The Pros and Cons of Working with WAB Widgets
```
Comment (atribolet) [inline _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: Is this heading accurate? Does this work?

File: _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md [hunk 2]
```diff
@@ -9,18 +9,27 @@ categories:
...
+But before we get to those tips, let's talk about some of the general pros and cons of working with WAB that I have discovered through this project. First, being able to focus on a single widget and not having to worry about an entire application has been great. I have loved letting Esri take care of all of the basic application stuff, such as loading the application framework, web map, and widgets. However, not having control over the entire system has created some headaches. That's where the WAB solutions came in.
+
+## Tips for Building WAB Widgets
```
Comment (atribolet) [inline _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: Again, I'm thinking subheadings with the specific tips might make the article more easily digestible at a glance? What are your thoughts on this @stdavis ?

File: _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md [hunk 3]
```diff
@@ -9,18 +9,27 @@ categories:
...
-* grunt tasks for JS tooling including babel, livereload, and sass
-* a grunt task to automate the copying of your development code to WAB developer edition
+* a best practice folder structure;
+* grunt tasks for JS tooling, including babel, livereload, and sass; and
+* a grunt task to automate the copying of your development code to WAB developer edition.
 
 Esri also includes a generator for scaffolding out the widgets themselves. This project provided a great starting point for my work.
 
-Another great tip that I found was to write your WAB widgets as plain old dojo widgets (inherit from `dijit._WidgetBase`) and then wrap them for WAB. This made unit tests much easier. The other benefit to this approach is that it makes it much easier to reuse the widgets in non-WAB applications. Here's an example repo: [gbochenek/wab-test-example](https://github.com/gbochenek/wab-test-example).
+### 2. Write your WAB widgets as dojo widgets.
```
Comment (atribolet) [inline _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: Again, please just double check the subheadings and whether they're even accurate! ;) And if we don't like them, that's ok! I thought the subheadings were worth exploring though.

File: _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md [hunk 4]
```diff
@@ -9,18 +9,27 @@ categories:
...
+### 3.Use Polyfill.io.
+Using babel worked great for this project until I needed polyfills for older browsers. (I'm looking at you, Internet Explorer.) Loading third-party libraries via AMD from within WAB widgets [is a pain](https://developers.arcgis.com/web-appbuilder/sample-code/add-a-third-party-library.htm). Fortunately, I found [Polyfill.io](https://polyfill.io/v3/). This is a service that dynamically returns a set of polyfills as defined by query parameters. The cool part about this service is that it only loads the code required for the browser that is making the request. Using this library from within WAB requires [adding it to the `loadResource` method in `init.js`](https://github.com/agrc/wfrc-wab-widgets#add-a-polyfill-for-older-browsers).
 
 I hope that these tips save you some headaches in the future. {% capture contact %}{% include contact.html
```
Comment (atribolet) [inline _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: Conclusions/wrapping things up are always so difficult. Bleh. But I feel like maybe there's room for just a little more easing into the contact info here at the end? Can we add something to kind of sum things up and transition into the closing more smoothly?

**Comment by stdavis** [comment]
This all looks great to me. The headers are great. I'll see if I can come up with more of a conclusion...

**Pull request merged**